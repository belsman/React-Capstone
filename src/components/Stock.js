import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const Stock = ({ stock }) => (
  <div className="card stock-card mb-1">
    <article className="card-body">
      <h5>{stock.ticker}</h5>
      <h6 className="text-truncate">{stock.companyName}</h6>
      <div className="my-2 py-2">
        <span className="">Price</span>
        <div className="d-flex justify-content-between">
          <span className="h5 text-secondary">
            $
            {' '}
            {stock.price}
          </span>
          <span className={stock.changes < 0 ? 'text-danger' : 'text-success'}>
            {stock.changesPercentage.replace(/[()]+/g, '')}
            {' '}
            <i className={stock.changes < 0 ? 'fa fa-arrow-down' : 'fa fa-arrow-up'} />
          </span>
        </div>
      </div>
      <Link to={`/stocks/${stock.ticker}`} className="btn btn-outline-info">
        View stock
      </Link>
    </article>
  </div>
);

Stock.propTypes = {
  stock: PropTypes.shape({
    ticker: PropTypes.string.isRequired,
    price: PropTypes.string.isRequired,
    changes: PropTypes.number.isRequired,
    changesPercentage: PropTypes.string.isRequired,
    companyName: PropTypes.string.isRequired,
  }).isRequired,
};

export default Stock;
