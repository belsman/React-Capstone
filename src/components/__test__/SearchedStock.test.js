import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import pretty from 'pretty';

/* eslint-disable no-unused-vars */
import { Link } from 'react-router-dom';
import Stock from '../SearchedStock';

jest.mock('react-router-dom', () => ({
  Link() {
    return (
      <div>A Dummy Link</div>
    );
  },
}));

let container = null;
beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

const searchedStock = {
  symbol: 'PRAA',
  name: 'PRA Group, Inc.',
  currency: 'USD',
  stockExchange: 'NasdaqGS',
  exchangeShortName: 'NASDAQ',
};

it('renders with with the correct stock info for searched stock', () => {
  act(() => {
    render(<Stock stock={searchedStock} />, container);
  });

  const h4 = document.querySelector('article h4');
  expect(h4.textContent).toBe('PRAA');

  const h6 = document.querySelector('article h6');
  expect(h6.textContent).toBe('PRA Group, Inc.');
});

it('should render the expected output', () => {
  act(() => {
    render(<Stock stock={searchedStock} />, container);
  });

  expect(
    pretty(container.innerHTML),
  ).toMatchSnapshot();
});
