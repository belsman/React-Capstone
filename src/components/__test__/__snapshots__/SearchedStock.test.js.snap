// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the expected output 1`] = `
"<div class=\\"card stock-card mb-1\\">
  <article class=\\"card-body\\">
    <h4>PRAA</h4>
    <h6 class=\\"text-truncate\\">PRA Group, Inc.</h6>
    <div class=\\"my-3 py-2\\">
      <div class=\\"d-flex justify-content-between mb-1\\"><span>Currency</span><span class=\\"h6 text-secondary\\">USD</span></div>
      <div class=\\"d-flex justify-content-between\\"><span>Exchange</span><span class=\\"h6 text-secondary\\">NasdaqGS</span></div>
    </div>
    <div>A Dummy Link</div>
  </article>
</div>"
`;
